### DISCUSSION TITLE
make minor modifications to the pdfs generated?

### DISCUSSION THREAD

**Discussion by zenny** [jgm/pandoc#10400] [Q&A]
Hi @jgm,

Great package that replaces most of other file format conversion tools! +1

I am using pandoc both from command line, and also using ox-pandoc emacs package (I have raised two issues, reportedly assuming that it has to do with eihther with emacs configurations (https://github.com/jkitchin/scimax/issues/509) or with the ox-pandoc package ( https://github.com/emacsorphanage/ox-pandoc/issues/43) which does not seem to be the case once I have started exploring a bit further as suggested by the developers in the related threads.

I am trying to achieve four things with pandoc because they are bottlenecks to the workflow of any scribe (discussed in https://github.com/pprevos/emacs-writing-studio/issues/24) :

1. Change the font from the default pandoc template while exporting from the orgfile or any other input files pandoc supports.
2. Insert a table of contents on a separate page from the title page and the rest of the contents ( tried appending `#+pandoc-options:` or `latex-header:` inside the org file, but did not work).
3. Specify LaTeX template (like `kaobook` (https://github.com/fmarotta/kaobook) ) either on orgmode file header or on command line to make the output file look as specified when exported to pdf.
4. Using `wkhtmltopdf` does a fantastic job to wrap the tables which is nearly impossible in orgmode without complex LaTeX configurationy, but generates a redundant horizontal scroll bar as if it is an webpage, not in pdfs. 

I appreciate if there are ways to overcome the bottlenecks as above. Thanking you in anticipation,

Cheers,
/zenny

**Comment by jgm** [comment]
1. Changing the font (for PDF output) just requires setting the `mainfont` as a variable or in metadata (assuming you're using `--pdf-engine=xelatex` which is recommended).  See the manual under "Variables for LaTeX".
2. You could probably achieve this by beginning your document with `\newpage`.
3. `kaobook` is a class not a template.  You could try setting `documentclass` and `classoption` variables.  Not sure how well it would work; it depends on the class.
4. Surprised to hear this. `wkhtmltopdf` is deprecated anyway, try `weasyprint` or `pagejs-cli`.  You could also try fiddling with `--pdf-engine-opt` to pass options to `wkhtmltopdf` (consult its manual).

**Reply by zenny** [reply]
Hi again @jgm:

Thanks for taking time to reply promptly. 

1. I tried specifying the specific font in the org file `#+LaTeX_Header: \usepackage{junicode}`, but pandoc export to pdf did not comply.
2. I tried with `\newpage` in org document, but had no effect. I have tried by inserting some LaTeX options as of below, but it did not generate TOC at all!
```
#+OPTIONS: timestamp:nil toc:nil
#+BEGIN_EXPORT latex
   \clearpage \setcounter{tocdepth}{5} \tableofcontents \clearpage
#+END_EXPORT
```
3.  Thanks for throwing light on the class stuff. I shall try to learn a bit more about it. Actually, I want to do everything from orgfile without touching the TeX file to make the workflow simple, fyi.
4. Glad that you explained about the pdf-rendering engines. I tried with `weasyprint`, but the rendering remained the same. If I rendered to pdf using from `-f org` to `-t html5` with in `--standalone` mode, some tables are rendered alright, but some tables are rendered (the table at the bottom in the screenshot below) with last column contents cropped (the one at the top in the screenshot below), meaning rendering is not consistent, as seen in the screenshot below:
![pic-selected-241123-0744-47](https://github.com/user-attachments/assets/c7e51499-f937-41db-8def-3505341104b0)

I appreciate if pandoc veterans here could hint me to overcome the issues as I have explained above. Have a nice weekend.

Cheers,
/zenny

**Reply by zenny** [reply]
An update:

1 (font issue) and 4 (table crop issue) above got resolved by appending the following:

```
#+PANDOC_OPTIONS: standalone:t table-of-contents:true toc-depth:5 pdf-engine:weasyprint number-sections:nil
#+PANDOC_VARIABLES: classoption:10pt,conference mainfont:Junicode sansfont:Monserrat monofont:FiraCodeMono mathfont:TeXGyreDejaVuMath
```
With help from:
1. https://github.com/kawabata/ox-pandoc/blob/aa37dc7e94213d4ebedb85c384c1ba35007da18e/README.org?plain=1#L135
2. https://superuser.com/questions/1349187/pandoc-how-to-get-pagebreak-between-title-block-and-the-table-of-contents
3. https://tex.stackexchange.com/questions/234786/how-to-set-a-font-family-with-pandoc
4. https://pandoc.org/org.html


Yet, I could not separate cover page and table of contents from the remaining body of text in separate pages even with:
```
#+PANDOC_METADATA: include-before: - '`\newpage{}`{=latex}' 
``` 

as advised in https://superuser.com/a/1349432. Any input appreciated.

**Reply by iandol** [reply]
For precise control, edit the pandoc templates: https://pandoc.org/MANUAL.html#templates -- you can see them all online here: https://github.com/jgm/pandoc-templates/blob/master/default.latex

Though looking at the `default.latex` template where `include-before` is it seems in the right place and should be working: https://github.com/jgm/pandoc-templates/blob/master/default.latex#L65 -- are you sure the latex looks ok (i.e. check the latex output before it becomes a PDF, maybe org mode is causing escapes in the latex characters or soemthing?)

Also PrinceXML is by far the best HTML to PDF engine, though it is not free for commercial use. For PDF overall you should seriously consider Typst. It now does much of what LaTeX can do, all in one tiny 10MB download with orders of magnitude less cruft and complexity...

**Reply by iandol** [reply]
You probably have some problem with how `include-before` is converting the string from org mode into metadata; at least with YAML metadata something like:


```markdown
---
title: A Test
author: Jane Doe
include-before: '`\newpage{}`{=latex}'
---

# Sun

Some *text*.

# Moon

Some *text*.
```

The `\newpage` command is in the right place and causes a pagebreak before TOC, i.e. it works. But it could be that going from org mode to YAML or soemthing causes the `\n` of newpage to get escaped or something?
